apps/prairielearn/src/lib/file-paths.ts

```diff
@@ -61,15 +63,20 @@ export async function questionFilePath(
       course_id: question.course_id,
       directory: question.template_directory,
     };
-    const result = await sqldb.queryZeroOrOneRowAsync(sql.select_question, params);
-    if (result.rowCount === 0) {
+    const templateQuestion = await sqldb.queryOptionalRow(
+      sql.select_question,
+      params,
+      QuestionSchema,
+    );
+    if (templateQuestion === null) {
       throw new error.HttpStatusError(
         500,
         `Could not find template question "${question.template_directory}" from question "${question.directory}"`,
       );
     }
 
-    const templateQuestion = result.rows[0];
+    assert(templateQuestion.directory !== null, 'template_directory is required');
```

Comment:
The `path.join` call in questionFilePath would fail.

Reply:
Nit, `template_directory` in the error message has no relation to an actual variable name. I'll fix in a followup.